### PR TITLE
snap: add `snap list -a` to show all snaps (even inactive ones)

### PR DESCRIPTION
**Pull request by mvo5**
LP: #1637096

### CODE REVIEW

File: cmd/snap/cmd_list.go [hunk 1]
```diff
@@ -39,6 +39,8 @@ type cmdList struct {
 	Positional struct {
 		Snaps []string `positional-arg-name:"<snap>"`
 	} `positional-args:"yes"`
+
+	All bool `short:"a" long:"all"`
```
Comment (niemeyer) [inline cmd/snap/cmd_list.go]: Can we please use either "-a" or "--all", but not both?  I'm personally fine with either. Having both generally leaves me that feeling that we didn't stop to think which form was appropriate, and then delegate to every user the need to understand that they both mean the same.

File: client/packages.go [hunk 2]
```diff
@@ -91,10 +91,23 @@ type FindOptions struct {
...
-func (client *Client) List(names []string) ([]*Snap, error) {
-	snaps, _, err := client.snapsFromPath("/v2/snaps", nil)
+func (client *Client) List(names []string, opts *ListOptions) ([]*Snap, error) {
+	if opts == nil {
+		opts = &ListOptions{}
+	}
+
+	q := make(url.Values)
+	if opts.All {
+		q.Add("all", "1")
```
Comment (niemeyer) [inline client/packages.go]: Looking at the implementation of /v2/find, we have "select=refresh" and "select=private" there, for example, which change the selection of snaps the operation applies to. If we want to follow suit, that might be "select=all", and defaults to "enabled" (implied when empty)?

Alternatively, if select was a bad plan we might sunset it at some point and move it to individual flags instead. But it'd be nice to have lookalike APIs on those two endpoints.

File: cmd/snap/cmd_list.go (hunk 1 above)
Comment (mvo5) [inline cmd/snap/cmd_list.go]: Thanks, I switch to `--all` only as its explicit and still short.

File: client/packages.go (hunk 2 above)
Comment (mvo5) [inline client/packages.go]: Indeed, `select={all,enabled}` feel nice (and more consistent with find). I updated the code but I'm keen to hear the opinion of @chipaca as well.

**Pull request merged**